tinygrad/tensor.py

```diff
@@ -521,11 +521,15 @@ def batchnorm(self, weight:Optional[Tensor], bias:Optional[Tensor], mean:Tensor,
     ret = x.mul(invstd.reshape(shape=[1, -1, 1, 1]) if len(invstd.shape) == 1 else invstd)
     return (ret + bias.reshape(shape=[1, -1, 1, 1])) if bias else ret
 
-  def dropout(self, p=0.5) -> Tensor:
-    if not Tensor.training: return self
+  def dropout(self, p=0.5, return_mask=False) -> Union[Tensor, Tuple[Tensor, Tensor]]:
+    if not Tensor.training:
+      # if mask is requested as output it will contain all True's.
+      return (self, Tensor.ones(*self.shape, requires_grad=False, dtype=dtypes.bool)) if return_mask else self
     # TODO: why is this going through numpy?
-    _mask: np.ndarray = np.asarray(Tensor._rng.binomial(1, 1.0-p, size=self.shape), dtype=np.float32)
-    return self * Tensor(_mask, requires_grad=False, device=self.device) * (1/(1.0 - p))
+    _mask: np.ndarray = Tensor._rng.choice(a=[False, True], size=self.shape, p=[p, 1-p])
```

Comment:
Hmm, is there a way to write this using the existing rng functions and not `Tensor._rng`

Reply:
might be related to int ops not being supported yet, the llvm ir codegen is using float functions and bool is using int

Reply:
Added a commit for this.

New onnx_ops code looks a bit ugly though because I had to write a wrapper for `RandomState`. This is because `RandomState.random()` doesn't accept a `dtype` param while `Generator.random()` does (and `Tensor.rand` uses it).

Plus, it slightly bothers me to replace `Tensor._rng` with an object that's not the type-hinted `np.random.Generator`.